marshmallow_dataclass/__init__.py

```diff
@@ -354,21 +356,27 @@ def class_schema(
         del current_frame
     _RECURSION_GUARD.seen_classes = {}
     try:
-        return _internal_class_schema(clazz, base_schema, clazz_frame)
+        return _internal_class_schema(clazz, base_schema, clazz_frame, None)
```

Comment:
have to add `None` here explicitly so we hit the cache for `_internal_class_schema`. lru_cache uses all given params as key, not adding the default values for not specified params